### PR TITLE
azure/container-registry: new module

### PR DESCRIPTION
**Pull request by samkahchiin**

### CODE REVIEW

File: azure/container-registry/variables.tf [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+variable "environment" {
+  type = string
+}
+
+variable "admin_enabled" {
+  type    = bool
+  default = true
+}
```
Comment (swiknaba) [inline azure/container-registry/variables.tf:24]: As discussed, this is fine for now. Can you please add an issue to the backlog to come back to this and refactor this to use either  a service principal or a managed identity? Thanks!

File: azure/container-registry/variables.tf [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+variable "public_network_access_enabled" {
+  type    = bool
+  default = true
+}
```
Comment (swiknaba) [inline azure/container-registry/variables.tf:29]: same here: good for now, but let's add an issue to disable public access, and instead use a private endpoint.

File: azure/container-registry/variables.tf [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  type        = number
+  default     = 7
+  description = "The number of days to retain an untagged manifest after which it gets purged."
+  nullable    = false
+}
+
+variable "sku" {
+  type        = string
+  description = "https://learn.microsoft.com/en-us/azure/container-registry/container-registry-skus"
+  default     = "Basic"
```
Comment (swiknaba) [inline azure/container-registry/variables.tf:60]: should we default to `"Standard"` to force the user to make a conscious decision for a less secure setup?

File: azure/container-registry/main.tf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      enabled            = true
+      key_vault_key_id   = var.key_vault_key_id
+      identity_client_id = var.encryption_client_id
+    }
+  }
+
+  tags = local.default_tags
+}
+
+# TODO: Research on the lifecycle policy
```
Comment (swiknaba) [inline azure/container-registry/main.tf:37]: great point! Please add an issue for this.

**Review by swiknaba — DISMISSED** [review]
LGTM! 🚀

File: azure/container-registry/variables.tf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+variable "admin_enabled" {
+  type    = bool
+  default = true
+}
+
+variable "public_network_access_enabled" {
+  type    = bool
+  default = true
```
Comment (samkahchiin) [inline azure/container-registry/variables.tf:28]:
```suggestion
  default = false
```

File: azure/container-registry/variables.tf (hunk 3 above)
Comment (samkahchiin) [inline azure/container-registry/variables.tf:60]:
```suggestion
  default     = "Standard"
```

**Pull request merged**